cloudinit/config/schema.py

```diff
@@ -740,6 +744,14 @@ def validate_cloudconfig_schema(
         ):
             # Schema was validated by netplan
             return True
+        elif netplan_available():
+            # We found no netplan API on netplan system, do not perform schema
+            # validation against cloud-init's network v2 schema because netplan
+            # supports more config keys than in cloud-init's netv2 schema.
+            # This may result in schema warnings for valid netplan config
+            # which would be successfully rendered by netplan but doesn't
+            # adhere to cloud-init's network v2.
+            return False
```

Comment:
Not for this PR, but this function's documentation is not immediately clear. If I read the documentation, without reading the code, I do not understand that the validation is materialized as log warnings and that the boolean return value represents if the validation was performed or not.

Reply:
> This change adds a new `False` exit point with a new meaning, invalidating the print statement in
> 
> https://github.com/canonical/cloud-init/blob/fdc1e45ce767a7e7a606662349ec1ea297017c3d/cloudinit/config/schema.py#L1177
> .

We don't actually get to 1177 when on netplan and no netplan API is present because of the new segment that I introduced 10 lines above that  https://github.com/canonical/cloud-init/blob/fdc1e45ce767a7e7a606662349ec1ea297017c3d/cloudinit/config/schema.py#L1157-L1162


> 
> ~At a minimum, could we extend that print message?~ This is not needed because `validate_cloudconfig_file` manually calls `netplan_validate_network_schema` and repeat this plumbing.

I do agree though that validate_cloudconfig_schema & _file functions need a significant refactor to avoid this complexity of schema handling. output formatting differences and duplication of logic. Maybe we take this as a separate issue to resolve at a later time?